rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/RxBleGattCallback.java

```diff
@@ -88,8 +69,17 @@ public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState
             super.onConnectionStateChange(gatt, status, newState);
             bluetoothGattProvider.updateBluetoothGatt(gatt);
 
-            propagateErrorIfOccurred(gattAndConnectionStateOutput, gatt, status, BleGattOperationType.CONNECTION_STATE);
-            gattAndConnectionStateOutput.valueRelay.call(new Pair<>(gatt, mapConnectionStateToRxBleConnectionStatus(newState)));
+            if (isDisconnectedOrDisconnecting(newState)) {
+                disconnectionRouter.route(new BleDisconnectedException(gatt.getDevice().getAddress()));
```

Comment:
Is there any reason to accept `BleException` here? I can see that only two options are possible here. Maybe there should be a method to report:

- BleDisconnected
- BleGattException

Reply:
One for each?
Do you think it would be good to add another `.route()` methods if different exceptions will be applicable?

Reply:
`BleException` is a general purpose exception. Form interface point of view it's not consistent with the class role - to route disconnections. I suggest names like "on ble disconnected", "handle gatt excption".